### ISSUE TITLE
Find the cause of slow server builds

### ISSUE THREAD

**Issue by menelike** [thereactivestack/meteor-webpack#181]
#### tl;dr we move all dev dependencies away from meteor, but make them accessible for `webpack` and therefore reduce the node_modules overhead

It's hard to crawl down the buildchain e.g. rabbit hole, but this what we've observed when using this **great** library with Meteor.

We updated to Meteor beta.1 to stop including devDependencies in productions builds. https://github.com/meteor/meteor/commit/7574f74481574222a494d0993f2b3286cfcdf4d6

We hoped that this might solve server builds as well. We were wrong.
https://github.com/meteor/meteor/issues/7008

It seems that Meteor just **touches** all node_modules on top level. When you start to use `webpack` you will end up using `babel` as well. This includes in a lot of presets, lot of files and a lot of used space.

Your `package.json` might look like this then

```
  "devDependencies": {
    "babel": "^6.3.26",
    "babel-core": "^6.3.26",
    "babel-eslint": "^5.0.0",
    "babel-loader": "^6.2.0",
    "babel-plugin-add-module-exports": "^0.1.2",
    "babel-plugin-react-transform": "^2.0.0",
    "babel-plugin-transform-decorators-legacy": "^1.3.2",
    "babel-polyfill": "^6.7.4",
    "babel-preset-es2015": "^6.3.13",
    "babel-preset-react": "^6.3.13",
    "babel-preset-stage-0": "^6.3.13",
    "babel-preset-stage-1": "^6.5.0",
    "css-loader": "^0.23.0",
    "eslint": "~2.2.0",
    "eslint-config-airbnb": "<6.2.0",
    "eslint-friendly-formatter": "^1.2.2",
    "eslint-plugin-react": "^4.2.3",
    "expose-loader": "^0.7.1",
    "extract-text-webpack-plugin": "^0.9.1",
    "faker": "^3.0.1",
    "file-loader": "^0.8.5",
    "json-loader": "^0.5.4",
    "react-transform-catch-errors": "^1.0.0",
    "react-transform-hmr": "^1.0.1",
    "redbox-react": "^1.2.0",
    "style-loader": "^0.13.0",
    "url-loader": "^0.5.7",
    "webpack": "^1.13.0",
    "webpack-hot-middleware": "^2.10.0"
  },
``` 

Most libs are needed for babel, eslint and webpack. All dev dependencies easily sum up to 450MB, production libs just consume about 50MB. Since Meteor uses NPMv2 this also means that a lot of subdependencies are included multiple times. This gets much better with NPMv3 as node_modules becomes flattened.

As a result, every server rebuild takes a hell of a time. We've tried a lot of tweaks, but in the end this is our current whacky solution to get productivity going on (dev only!).

1. We added `modulesDirectories` to webpack.settings and patched the code to consume it
2. We installed all productions libs in meteor with `meteor npm install --only=prod`
3. We installed all dev libs on PWD ./../ with `npm install --only=dev` **we use NPMv3 here!**
4. Now we need to tell webpack where the dev dependencies can be found and we set modulesDirectories to '../node_modules' in webpack.settings

*this might break some stuff, it highly depends on your project, e.g. eslint stopped working as the configuration would need to be adjusted as well*

Initial webpack builds drop from 52sec to 30 sek, server rebuilds are nearly twice as fast (highly depends on your hardware).

The reason why this works is that Meteor currently uses lot of disc IO. It seems that Meteor *touches?* everything in node_modules on rebuild.

We are waiting for Meteor v1.3.3 where babelrc support will be included. We might be able to exclude a lot of babel parts from dev dependencies. But this won't work as a proper solution.

Can anyone confirm this? Does anyone know the root of the problem or a better solution?

**Comment by eXon** [comment]
Meteor will need to fix this (they already know what that issue). There is not much this plugin can do more about it.
